### PR TITLE
Default histogram observer

### PR DESCRIPTION
**Pull request by raghuramank100**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26516 Per-channel baseline
* #26520 control of observer/fake-quant operations
* **#26622 Default histogram observer**
* #26519 Serialization and range reduction support for Fake Quant/Observer
* #26876 Fake quantization enhancements for QAT/PTQ support- fix tests

Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

### CODE REVIEW

File: torch/quantization/observer.py [hunk 1]
```diff
@@ -586,6 +586,11 @@ def default_observer(**kwargs):
     kwargs.setdefault("reduce_range", True)
     return observer(MinMaxObserver, **kwargs)
 
+def default_l2_observer(**kwargs):
```
Comment (jerryzh168) [inline torch/quantization/observer.py]: nit: should we use default_histogram_observer to align the names?

File: torch/quantization/observer.py [hunk 2]
```diff
@@ -586,6 +586,11 @@ def default_observer(**kwargs):
...
+    # Restrict activations to be in the range (0,127)
+    print(kwargs)
```
Comment (jerryzh168) [inline torch/quantization/observer.py]: remove

File: torch/quantization/observer.py [hunk 3]
```diff
@@ -605,8 +605,8 @@ def calculate_qparams(self):
     def get_tensor_value(self):
         return self.tensor_val
 
-
 # Restrict activations to be in the range (0,127)
 default_observer = MinMaxObserver.with_args(reduce_range=True)
 default_debug_observer = RecordingObserver
 default_weight_observer = MinMaxObserver.with_args(dtype=torch.qint8, qscheme=torch.per_tensor_symmetric)
+default_histogram_observer = HistogramObserver.with_args(reduce_range=True)
```
Comment (dzhulgakov) [inline torch/quantization/observer.py]: should it be "per architecture" defaults?

Comment (jerryzh168) [reply torch/quantization/observer.py]: do you mean having default_histogram_observer_fbgemm and default_histogram_observer_qnnpack?

Comment (raghuramank100) [reply torch/quantization/observer.py]: That comes in a later diff, where we specify an architecture to obtain a qconfig.
https://github.com/pytorch/pytorch/pull/26627

**Review by jerryzh168 — APPROVED** [review]
LG

**Pull request closed without merging**